src/solver/conv_hip_implicit_gemm_v4r4_gen_xdlops_wrw_fp32.cpp

```diff
@@ -0,0 +1,604 @@
+    const std::size_t wi = ConvolutionContextInterpreter::GetInputWidthWi(ctx);
+    const std::size_t conv_stride_h =
+        ConvolutionContextInterpreter::GetAdjustedConvolutionStrideH(ctx);
+    const std::size_t conv_stride_w =
+        ConvolutionContextInterpreter::GetAdjustedConvolutionStrideW(ctx);
+    const std::size_t conv_dilation_h =
+        ConvolutionContextInterpreter::GetAdjustedConvolutionDilationH(ctx);
+    const std::size_t conv_dilation_w =
+        ConvolutionContextInterpreter::GetAdjustedConvolutionDilationW(ctx);
+    const std::size_t in_left_pad_h = ConvolutionContextInterpreter::GetInputLeftPadH(ctx);
+    const std::size_t in_left_pad_w = ConvolutionContextInterpreter::GetInputLeftPadW(ctx);
+    const std::size_t in_right_pad_h =
+        ConvolutionContextInterpreter::GetAdjustedInputRightPadH(ctx);
+    const std::size_t in_right_pad_w =
+        ConvolutionContextInterpreter::GetAdjustedInputRightPadW(ctx);
+
+    const std::size_t GemmM = k;
+    const std::size_t GemmN = c * y * x;
+
+    const std::size_t block_size = config.GemmNPerBlock * config.GemmMPerBlock /
+                                   (config.GemmMPerWave * config.GemmNPerWave) * wave_size;
+    const std::size_t grid_size =
+        (GemmN / config.GemmNPerBlock) * (GemmM / config.GemmMPerBlock) * config.GemmKBlocks;
+
+    std::size_t lkl_wk0 = block_size;
+    std::size_t lkl_wk1 = 1;
+    std::size_t lkl_wk2 = 1;
+
+    construction_parameters.l_wk.push_back(lkl_wk0);
+    construction_parameters.l_wk.push_back(lkl_wk1);
+    construction_parameters.l_wk.push_back(lkl_wk2);
+
+    std::size_t gbl_wk0 = lkl_wk0 * grid_size;
+    std::size_t gbl_wk1 = 1;
+    std::size_t gbl_wk2 = 1;
+
+    construction_parameters.g_wk.push_back(gbl_wk0);
+    construction_parameters.g_wk.push_back(gbl_wk1);
+    construction_parameters.g_wk.push_back(gbl_wk2);
+
+    int GemmABlockCopyClusterLengths_GemmK  = 0;
+    int GemmABlockCopyClusterLengths_GemmM  = 0;
+    int GemmABlockCopySrcDataPerRead_GemmK  = 0;
+    int GemmABlockCopyDstDataPerWrite_GemmM = 0;
+    int GemmBBlockCopyClusterLengths_GemmK  = 0;
+    int GemmBBlockCopyClusterLengths_GemmN  = 0;
+    int GemmBBlockCopySrcDataPerRead_GemmK  = 0;
+    int GemmBBlockCopyDstDataPerWrite_GemmN = 0;
+
+    std::tie(GemmABlockCopyClusterLengths_GemmK,
+             GemmABlockCopyClusterLengths_GemmM,
+             GemmABlockCopySrcDataPerRead_GemmK,
+             GemmABlockCopyDstDataPerWrite_GemmM,
+             std::ignore) = config.CalculateGemmABlockCopyPerformanceParameters(ctx);
+
+    std::tie(GemmBBlockCopyClusterLengths_GemmK,
+             GemmBBlockCopyClusterLengths_GemmN,
+             GemmBBlockCopySrcDataPerRead_GemmK,
+             GemmBBlockCopyDstDataPerWrite_GemmN,
+             std::ignore) = config.CalculateGemmBBlockCopyPerformanceParameters(ctx);
+
+    construction_parameters.kernel_file = "gridwise_convolution_implicit_gemm_v4r4_gen_xdlops_wrw_"
```

Comment:
can you please not split this kernel name/file name into 2 separate lines? Searching for this string in grep makes it difficult. I believe, clang-format will be splitting into 2 for you. You could disable clang-format by 

// clang-format off

// clang-format on